lib/vmfloaty/utils.rb

```diff
@@ -45,7 +56,7 @@ def self.standardize_hostnames(response_body)
     end
 
     # vmpooler reports the domain separately from the hostname
-    domain = response_body.delete('domain')
+    domain = response_body.delete('domain') if response_body.has_key?('domain')
```

Comment:
the `if` is not needed since
response_body.delete('domain')
will return `nil` when that key does not exist.